purview/utils/general.py

```diff
@@ -151,20 +151,22 @@ def create_query(item, uid_name, uid, reverse=False):
     if curr_node_label not in story_flow:
         raise ValidationError('The story is not available for this kind of resource')
 
+    count = 0
     while story_flow[curr_node_label][rel_label]:
         if curr_node_label == item.__label__:
-            node = node_query(curr_node_label, uid_name, uid)
-        else:
-            node = node_query(curr_node_label)
+            query += 'MATCH {0}'.format(node_query(curr_node_label, uid_name.rstrip('_'), uid))
+            query += '\nCALL apoc.path.expandConfig({0}, {{sequence:\'{1}'.format(
+                curr_node_label.lower(), curr_node_label)
 
-        next_node = node_query(story_flow[curr_node_label][node_label])
-        query += 'OPTIONAL MATCH {0}-[:{1}]-{2}\n'.format(
-            node, story_flow[curr_node_label][rel_label], next_node)
+        query += ', {0}, {1}'.format(
+            story_flow[curr_node_label][rel_label], story_flow[curr_node_label][node_label])
 
+        count += 1
         curr_node_label = story_flow[curr_node_label][node_label]
 
     if query:
```

Comment:
Optional: It'd be easier to read if this was `if count:`

Reply:
We later discussed on removing the count, so I will keep this as it is :)